RecoLocalCalo/HcalRecProducers/src/ZdcHitReconstructor_Run3.cc

```diff
@@ -0,0 +1,153 @@
+#include "ZdcHitReconstructor_Run3.h"
+#include "DataFormats/Common/interface/EDCollection.h"
+#include "DataFormats/Common/interface/Handle.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "CalibFormats/HcalObjects/interface/HcalCoderDb.h"
+#include "CalibFormats/HcalObjects/interface/HcalCalibrations.h"
+
+#include "CalibFormats/HcalObjects/interface/HcalDbService.h"
+#include "CalibFormats/HcalObjects/interface/HcalDbRecord.h"
+#include "RecoLocalCalo/HcalRecAlgos/interface/HcalSeverityLevelComputer.h"
+#include "RecoLocalCalo/HcalRecAlgos/interface/HcalSeverityLevelComputerRcd.h"
+#include "Geometry/CaloTopology/interface/HcalTopology.h"
+
+#include <iostream>
+
+#include "CalibFormats/HcalObjects/interface/HcalCalibrationWidths.h"
+#include "CondFormats/HcalObjects/interface/HcalPedestal.h"
+
+namespace zdchelper {
+  void setZDCSaturation(ZDCRecHit rh, QIE10DataFrame& digi, int maxValue) {
+    unsigned int digisize = digi.size();
+    for (unsigned int i = 0; i < digisize; i++) {
```

Comment:
Use a range-based for loop

Reply:
I understand a range-base loop to have the format `for ( int i : {1,2,3})`. I tried to do that with QIE10Dataframe but compiling failed. I used to the same formatting used[ here](https://cmssdt.cern.ch/lxr/source/RecoLocalCalo/HcalRecAlgos/src/HcalADCSaturationFlag.cc) which was used previously but does not work with QIE10Dataframes.

Reply:
I'm not sure I understand your response.
Here's an example of a range-based for loop:
https://github.com/cms-sw/cmssw/blob/9cd76b9aeb39c940d3ad14bba2d25cf65f7ee23a/RecoLocalCalo/HcalRecProducers/src/ZdcHitReconstructor_Run3.cc#L103

Reply:
Fixed! My issue was with conversion to QIE10DataFrame::Sample but that has been resolved